phpBB/phpbb/install/module/obtain_data/task/obtain_server_data.php

```diff
@@ -0,0 +1,203 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+namespace phpbb\install\module\obtain_data\task;
+
+use phpbb\install\exception\user_interaction_required_exception;
+
+/**
+ * This class requests and saves some information about the server
+ */
+class obtain_server_data extends \phpbb\install\task_base implements \phpbb\install\task_interface
+{
+	/**
+	 * @var \phpbb\install\helper\config
+	 */
+	protected $install_config;
+
+	/**
+	 * @var \phpbb\install\helper\iohandler\iohandler_interface
+	 */
+	protected $io_handler;
+
+	/**
+	 * Constructor
+	 *
+	 * @param \phpbb\install\helper\config							$config		Installer's config
+	 * @param \phpbb\install\helper\iohandler\iohandler_interface	$iohandler	Installer's input-output handler
+	 */
+	public function __construct(\phpbb\install\helper\config $config,
+								\phpbb\install\helper\iohandler\iohandler_interface $iohandler)
+	{
+		$this->install_config	= $config;
+		$this->io_handler		= $iohandler;
+
+		parent::__construct(true);
+	}
+	/**
+	 * {@inheritdoc}
+	 */
+	public function run()
+	{
+		$cookie_secure = $this->io_handler->is_secure();
+		$server_protocol = ($this->io_handler->is_secure()) ? 'https://' : 'http://';
+		$server_port = $this->io_handler->get_server_variable('SERVER_PORT', 0);
+
+		// HTTP_HOST is having the correct browser url in most cases...
+		$server_name = strtolower(htmlspecialchars_decode($this->io_handler->get_header_variable(
+			'Host',
+			$this->io_handler->get_server_variable('SERVER_NAME')
+		)));
+
+		// HTTP HOST can carry a port number...
+		if (strpos($server_name, ':') !== false)
+		{
+			$server_name = substr($server_name, 0, strpos($server_name, ':'));
+		}
+
+		$script_path = htmlspecialchars_decode($this->io_handler->get_server_variable('PHP_SELF'));
+
+		if (!$script_path)
+		{
+			$script_path = htmlspecialchars_decode($this->io_handler->get_server_variable('REQUEST_URI'));
+		}
+
+		$script_path = str_replace(array('\\', '//'), '/', $script_path);
+		$script_path = trim(dirname(dirname(dirname($script_path)))); // Because we are in install/app.php/route_name
```

Comment:
we should use the Symfony Request object to retrieve the default values. 
But it can be done in another PR. I'm just writing it for the record.